src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -68,18 +68,27 @@ default String getAsTextHidePrivate() {
         final StringBuilder builder = new StringBuilder();
         builder.append(getName());
         if (!getPhone().isPrivate()) {
-            builder.append(" Phone: ").append(getPhone());
+            builder.append(" " + getPrintableString(getPhone()));
         }
         if (!getEmail().isPrivate()) {
-            builder.append(" Email: ").append(getEmail());
+            builder.append(" " + getPrintableString(getEmail()));
         }
         if (!getAddress().isPrivate()) {
-            builder.append(" Address: ").append(getAddress());
+            builder.append(" " + getPrintableString(getAddress()));
         }
         builder.append(" Tags: ");
         for (Tag tag : getTags()) {
             builder.append(tag);
         }
         return builder.toString();
     }
+
+    default String getPrintableString(Printable... printables){
+        String str = "";
+        for(Printable printable : printables){
+            str += printable.getPrintableString();
```

Comment:
Wouldn't this result in an ugly string with no separator between each part? 😝

Reply:
:bulb: *Java tip*: Java 8 has a `StringJoiner` class that can join multiple strings using a delimiter e.g. `aaa` `bbb` `ccc` --> `aaa, bbb, ccc` https://www.mkyong.com/java8/java-8-stringjoiner-example/